x-pack/plugins/cases/server/client/configure/client.ts

```diff
@@ -364,10 +407,59 @@ export async function create(
     const validatedConfigurationRequest =
       decodeWithExcessOrThrow(ConfigurationRequestRt)(configRequest);
 
-    validateDuplicatedCustomFieldKeysInRequest({
-      requestCustomFields: validatedConfigurationRequest.customFields,
+    validateDuplicatedKeysInRequest({
+      requestFields: validatedConfigurationRequest.customFields,
+      fieldName: 'customFields',
+    });
+
+    validateDuplicatedKeysInRequest({
+      requestFields: validatedConfigurationRequest.templates,
+      fieldName: 'templates',
     });
 
+    if (validatedConfigurationRequest.templates && validatedConfigurationRequest.templates.length) {
+      const hasPlatinumLicenseOrGreater = await licensingService.isAtLeastPlatinum();
+      const hasCustomFieldsInConfiguration =
+        validatedConfigurationRequest.customFields &&
+        validatedConfigurationRequest.customFields.length > 0;
+
+      validatedConfigurationRequest.templates.forEach((template, index) => {
+        if (!hasCustomFieldsInConfiguration && template.caseFields?.customFields?.length) {
+          throw Boom.badRequest(
+            'Cannot create template with custom fields as there are no custom fields in configuration'
+          );
+        }
+
+        /**
+         * validate template's custom fields keys
+         */
+        validateDuplicatedKeysInRequest({
+          requestFields: template?.caseFields?.customFields,
+          fieldName: `templates[${index}]'s customFields`,
+        });
+
+        /**
+         * Assign users to a template is only available to Platinum+
+         */
+
+        const hasAssigneesInTemplate =
+          template.caseFields?.assignees && template.caseFields?.assignees.length > 0;
+
+        if (hasAssigneesInTemplate && !hasPlatinumLicenseOrGreater) {
+          throw Boom.forbidden(
+            'In order to assign users to cases, you must be subscribed to an Elastic Platinum license'
+          );
+        }
+
+        licensingService.notifyUsage(LICENSING_CASE_ASSIGNMENT_FEATURE);
+      });
+
+      validateTemplatesCustomFieldsInRequest({
+        templates: validatedConfigurationRequest.templates,
+        customFieldsConfiguration: validatedConfigurationRequest.customFields,
+      });
+    }
+
```

Comment:
How about doing the same as in `update`? The logic seems similar.

```suggestion
if (validatedConfigurationRequest.templates && validatedConfigurationRequest.templates.length) {
  const hasAssigneesInTemplate = validatedConfigurationRequest.templates.find(
    (template) => template.caseFields?.assignees && template.caseFields?.assignees.length > 0
  );

  const hasPlatinumLicenseOrGreater = await licensingService.isAtLeastPlatinum();

  if (hasAssigneesInTemplate && !hasPlatinumLicenseOrGreater) {
    throw Boom.forbidden(
      'In order to assign users to cases, you must be subscribed to an Elastic Platinum license'
    );
  }

  licensingService.notifyUsage(LICENSING_CASE_ASSIGNMENT_FEATURE);
}

validateTemplatesCustomFieldsInRequest({
  templates: validatedConfigurationRequest.templates,
  customFieldsConfiguration: validatedConfigurationRequest.customFields,
});
```

(I didn't 100% check the code but I mean something like this)

Reply:
yeah create and update both have validations, I will move it to common function

Reply:
fixed